src/Provider/FileProvider.php

```diff
@@ -257,13 +259,17 @@ public function getDownloadResponse(MediaInterface $media, string $format, strin
             throw new \RuntimeException(sprintf('Cannot use X-Sendfile or X-Accel-Redirect with non %s.', Local::class));
         }
 
-        $filename = sprintf(
-            '%s/%s',
-            $adapter->getDirectory(),
-            $this->generatePrivateUrl($media, $format)
-        );
+        $directory = $adapter->getDirectory();
+
+        if (false === $directory) {
+            throw new \RuntimeException('Cannot retrieve directory from the adapter.');
```

Comment:
I did throw an exception here, it is correct?

Reply:
Seems right to me, that's the strategy I generally use